docs/migration-2.0.md

```diff
@@ -0,0 +1,1178 @@
+# Migrating from 1.x to 2.0
+
+The minimum supported version of Go for v2 is 1.18.
+
+To upgrade imports of the Go Driver from v1 to v2, we recommend using [marwan-at-work/mod
+](https://github.com/marwan-at-work/mod):
+
+```
+mod upgrade --mod-name=go.mongodb.org/mongo-driver
+```
+
+## Description Package
+
+The `description` package has been removed in v2.
+
+## Event Package
+
+References to `description.Server` and `description.Topology` have been replaced with `event.ServerDescription` and `event.TopologyDescription`, respectively. Additionally, the following changes have been made to the fields:
+
+- `Kind` has been changed from `uint32` to `string` for ease of use.
+- `SessionTimeoutMinutes` has been changed from `uint32` to `*int64` to differentiate between a zero timeout and no timeout.
+
+The following event constants have been renamed to match their string literal value:
+
+- `PoolCreated` to `ConnectionPoolCreated`
+- `PoolReady` to `ConnectionPoolRead`
+- `PoolCleared` to `ConnectionPoolCleared`
+- `PoolClosedEvent` to `ConnectionPoolClosed`
+- `GetStarted` to `ConnectionCheckOutStarted`
+- `GetFailed` to `ConnectionCheckOutFailed`
+- `GetSucceeded` to `ConnectionCheckedOut`
+- `ConnectionReturned` to `ConnectionCheckedIn`
+
+### CommandFailedEvent
+
+`CommandFailedEvent.Failure` has been converted from a string type to an error type to convey additional error information and to match the type of other `*.Failure` fields, like `ServerHeartbeatFailedEvent.Failure`.
+
+### CommandFinishedEvent
+
+The type for `ServerConnectionID` has been changed to `*int64` to prevent an `int32` overflow and to differentiate between an ID of `0` and no ID.
+
+### CommandStartedEvent
+
+The type for `ServerConnectionID` has been changed to `*int64` to prevent an `int32` overflow and to differentiate between an ID of `0` and no ID.
+
+### ServerHeartbeatFailedEvent
+
+`DurationNanos` has been removed in favor of `Duration`.
+
+### ServerHeartbeatSucceededEvent
+
+`DurationNanos` has been removed in favor of `Duration`.
+
+## Mongo Package
+
+### Client
+
+#### Connect
+
+`Client.Connect` has been removed in favor of `mongo.Connect()`. See section on `NewClient` for more details.
+
+The `context.Context` parameter has been removed from `mongo.Connect()` since [deployment connector ](https://github.com/mongodb/mongo-go-driver/blob/a76687682f080c9612295b646a00650d00dd16e1/x/mongo/driver/driver.go#L91) doesn’t accept a context, so the context passed to `mongo.Connect` doesn’t actually do anything.
```

Comment:
Extra space in [deployment connector ]
```suggestion
The `context.Context` parameter has been removed from `mongo.Connect()` since [deployment connector](https://github.com/mongodb/mongo-go-driver/blob/a76687682f080c9612295b646a00650d00dd16e1/x/mongo/driver/driver.go#L91) doesn’t accept a context, so the context passed to `mongo.Connect` doesn’t actually do anything.
```